pkg/controllers/nodetemplate/controller.go

```diff
@@ -96,17 +112,35 @@ func (c *Controller) resolveSubnets(ctx context.Context, nodeTemplate *v1alpha1.
 	return nil
 }
 
-func (c *Controller) resolveSecurityGroup(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
+func (c *Controller) resolveSecurityGroups(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
 	securityGroupIds, err := c.securityGroupProvider.List(ctx, nodeTemplate)
 	if err != nil {
 		return err
 	}
 
-	nodeTemplate.Status.SecurityGroups = lo.Map(securityGroupIds, func(id string, _ int) v1alpha1.SecurityGroupStatus {
-		return v1alpha1.SecurityGroupStatus{
+	nodeTemplate.Status.SecurityGroups = lo.Map(securityGroupIds, func(id string, _ int) v1alpha1.SecurityGroup {
+		return v1alpha1.SecurityGroup{
 			ID: id,
 		}
 	})
 
 	return nil
 }
+
+func (c *Controller) resolveAMIs(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
+	amiRequirement, err := c.amiProvider.Get(ctx, nodeTemplate, &amifamily.Options{})
+	if err != nil {
+		nodeTemplate.Status.AMIs = nil
```

Comment:
What happens if this error is a transient error? I don't think we should clear this whenever we get an error. The correct behavior seems like clearing it if we get no resolved amis and also firing an error for resolving no amis